src/utils/azure_storage.ts

```diff
@@ -317,7 +318,7 @@ export async function retrieveTableEntity<A, O, I>(
   options: azureStorage.TableService.TableEntityRequestOptions = {
     entityResolver: getValueOnlyEntityResolver
   }
-): Promise<Either<Error, Option<A>>> {
+): Promise<Either<Error | ReadonlyArray<t.ValidationError>, Option<A>>> {
```

Comment:
interpreting `Error | ReadonlyArray<t.ValidationError>` will require runtime checks - we should stick to either one

one option would be that, since `Error` is not well typed anyway, to just reject the promise with the error in case of non-404 errors

another option would be to return a nested `Either`:  `Promise<Either<Either<Error, ReadonlyArray<t.ValidationError>>, Option<A>>>`

however, when return types become so complicated, it's usually a sign that the function is trying to do too many things - it could be made simpler and rely on composing multiple functions - if we remove the entity decoding from this function, we can have it return a:

`Promise<Either<StorageError, Option<unknown>>>`

then pipe the result into the io-ts decode with a map

Reply:
Done